### ISSUE TITLE
Add support for `/cygdrive` paths returned by some versions of Cygwin

### ISSUE THREAD

**Issue by jantajov** [microsoft/vscode-cpptools#10112]
Hi,

I am facing error: "Cannot open source file stddef.h" (dependency of stdlib.h)" when editing c files.

![image](https://user-images.githubusercontent.com/12242398/200069265-eb2d4a9d-25bc-44b0-b553-f137d6f1592a.png)

Archive with CygWin installation: https://1drv.ms/u/s!AhQbYFkanIg5h5RrrKO2FXw8tPj23Q?e=1igDaK

Config: 
```JSON
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": ["${workspaceFolder}/**"],
            "defines": ["_DEBUG", "UNICODE", "_UNICODE"],
            "compilerPath": "C:\\Program Files\\CygWin64\\bin\\gcc.exe",
            "cStandard": "c99",
            "cppStandard": "c++17",
            "intelliSenseMode": "linux-gcc-x64"
        }
    ],
    "version": 4
}
```

Weirdly enough Intelli-Sense seems to work just fine.
![image](https://user-images.githubusercontent.com/12242398/200069169-98e45f06-747f-4adc-a372-47de547497c5.png)

![image](https://user-images.githubusercontent.com/12242398/200069109-f16edb36-f4e5-4809-90bf-0a46da2d880a.png)

**Comment by sean-mcmanus** [comment]
You may not have the required headers/libraries installed. Are you able to compile with C:\\Program Files\\CygWin64\\bin\\gcc.exe ?

What output do you get when you run the C/C++: Log Diagnostics command from the Command Palette? The includePath to stddef.h is most likely missing. You might try searching for that file.

**Comment by jantajov** [comment]
Hi Sean,

> You may not have the required headers/libraries installed. Are you able to compile with C:\Program Files\CygWin64\bin\gcc.exe ?

If I press Ctrl + Shift + B it builds just fine. Giving me folowing output:

```
 *  Executing task: C/C++: gcc.exe build active file 

Starting build...
"C:\Program Files\CygWin64\bin\gcc.exe" -fdiagnostics-color=always -g C:\Users\tajovsky\Desktop\c-proj\main.c -o C:\Users\tajovsky\Desktop\c-proj\main.exe

Build finished successfully.
 *  Terminal will be reused by tasks, press any key to close it. 
```

here is output from Log Diagnostics. LGTM...

```
-------- Diagnostics - 4. 11. 2022 23:57:47
Version: 1.12.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "compilerPath": "C:\\Program Files\\CygWin64\\bin\\gcc.exe",
    "cStandard": "c99",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-gcc-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Users\tajovsky\Desktop\c-proj\main.c ]:
    C:\Users\tajovsky\Desktop\c-proj\main.c
Translation Unit Configurations:
[ C:\Users\tajovsky\Desktop\c-proj\main.c ]:
    Process ID: 7776
    Memory Usage: 17 MB
    Compiler Path: c:\program files\cygwin64\bin\gcc.exe
    Includes:
        C:\PROGRAM FILES\CYGWIN64\USR\INCLUDE
        C:\PROGRAM FILES\CYGWIN64\USR\INCLUDE\W32API
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: c99
    IntelliSense Mode: linux-gcc-x64
    Other Flags:
        --gcc
        --gnu_version=110300
Total Memory Usage: 17 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 1843
```


>Also, you may try changing your intelliSenseMode to "windows-gcc-x64" to see if that helps.

Changing intelliSenseMode and restarting VSCode didn't help.

**Comment by sean-mcmanus** [comment]
Can you do a search under C:\PROGRAM FILES\CYGWIN64\ for a file named stddef.h?

**Comment by jantajov** [comment]
Sure no problem:

C:\Program Files\CygWin64>bin\find.exe -name stddef.h
./lib/gcc/x86_64-pc-cygwin/11/include/stddef.h
./usr/lib/gcc/x86_64-pc-cygwin/11/include/stddef.h

**Comment by sean-mcmanus** [comment]
For some reason, our compiler query is not returning `C:\Program Files\CygWin64\lib\gcc\x86_64-pc-cygwin\11\include`. Are you able to fix the issue via adding that path to the includePath?
```
    "includePath": [
        "${workspaceFolder}/**",
        "C:\\Program Files\\CygWin64\\lib\\gcc\\x86_64-pc-cygwin\\11\\include"
    ],
```

**Comment by jantajov** [comment]
Well, yes it works.
**But...**
That isn't a real fix for the issue. There might be more paths that need to be included. Why is VS Code compiler query failing to find all inclusion paths?

Are you able to reproduce the issue on your local machine, so you could take a closer look?

In my original comment I have linked the archive with the whole CygWin environment. Perhaps you could spin up Windows Sandbox, install VSCode there and unpack my CygWin archive in C:\Program Files\CygWin64.

**Comment by Colengms** [comment]
It looks like this version of Cygwin returns include (full) paths under a `/cygdrive` path, which I've never seen before.

```
c:\Program Files\CygWin64\bin>"c:\program files\cygwin64\bin\gcc.exe" -std=c++17 -m64 -Wp,-v -E -dM -x c++ /dev/null
...
 /cygdrive/c/program files/cygwin64/bin/../lib/gcc/x86_64-pc-cygwin/11/include/c++
 /cygdrive/c/program files/cygwin64/bin/../lib/gcc/x86_64-pc-cygwin/11/include/c++/x86_64-pc-cygwin
 /cygdrive/c/program files/cygwin64/bin/../lib/gcc/x86_64-pc-cygwin/11/include/c++/backward
 /cygdrive/c/program files/cygwin64/bin/../lib/gcc/x86_64-pc-cygwin/11/include
 /usr/include
 /usr/lib/../lib/../include/w32api
...
```

Using this issue to track adding support for full paths under `/cygdrive`.

**Comment by sean-mcmanus** [comment]
Fixed with 1.13.4 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.4
